src/util/ConfigManager/ConfigManager.h

```diff
@@ -186,10 +194,78 @@ class ConfigManager {
   */
   std::string printConfigurationDoc(bool printCurrentJsonConfiguration) const;
 
+  /*
+  @brief Add a function to the configuration manager for verifying, that the
+  values of the given configuration options are valid. Will always be called
+  after parsing.
+
+  @tparam ValidatorParameterTypes The types of the values in the
+  `validatorParameter`. Can be left up to type deduction.
+
+  @param validatorFunction Checks, if the values of the given configuration
+  options are valid. Should return true, if they are valid.
+  @param errorMessage A `std::runtime_error` with this as an error message will
+  get thrown, if the `validatorFunction` returns false.
+  @param validatorParameter Proxies for the configuration options, whos values
+  will be passed to the validator function as function arguments. Will keep the
+  same order.
+  */
+  template <typename... ValidatorParameterTypes>
+  void addValidator(
+      Validator<ValidatorParameterTypes...> auto validatorFunction,
+      std::string_view errorMessage,
+      ConfigOptionProxy<ValidatorParameterTypes>... validatorParameter)
+      requires(sizeof...(validatorParameter) > 0 &&
+               sizeof...(ValidatorParameterTypes) ==
+                   sizeof...(validatorParameter)) {
```

Comment:
Isn't the second check redundant?

Reply:
You are right. Fixed in [2f9c11c](https://github.com/ad-freiburg/qlever/pull/1058/commits/2f9c11c2e77051adb911d528ce196d5dcb4c848d).